### PR TITLE
Rename pointwise ops

### PR DESCRIPTION
**Pull request by kahmed10**
For clarity on kernel names found when profiling. The new names are set to the order of the ops being compiled. For example: add + relu = add_relu_kernel.

### CODE REVIEW

**Comment by pfultz2** [comment]
This should be based on #1125.

**Comment by pfultz2** [comment]
I think it would be better to just have a function to iterator over operators rather than do this in the generator:

```cpp
std::vector<std::string> get_op_names(module& m)
{
    std::vector<std::string> result;
    for(auto& ins:m)
    {
        if (starts_with(ins->name(), "@"))
            continue;
        result.push_back(ins->name());
    }
    return result;
}
```

**Comment by causten** [comment]
we are doing more profiling lately.  I feel this will help.  You should push to get this in

**Comment by kahmed10** [comment]
> we are doing more profiling lately. I feel this will help. You should push to get this in

Noted. Will prioritize and open for review soon.

**Comment by kahmed10** [comment]
> I think it would be better to just have a function to iterator over operators in a module rather than do this in the generator:
> 
> ```c++
> std::vector<std::string> get_op_names(module& m)
> {
>     std::vector<std::string> result;
>     for(auto& ins:m)
>     {
>         if (starts_with(ins->name(), "@"))
>             continue;
>         result.push_back(ins->name());
>     }
>     return result;
> }
> ```

changed

File: src/targets/gpu/jit/pointwise.cpp [hunk 1]
```diff
@@ -131,12 +143,20 @@ struct pointwise_compiler : compiler<pointwise_compiler>
...
+        else
+            op_name_string += "_kernel";
+        options.kernel_name = op_name_string;
         options.set_launch_params(v,
                                   compute_global_for(ctx,
                                                      options.output.elements() / vec_size,
                                                      oversubscribe_if(not is_preloading)));
         auto src = interpolate_string(pointwise_kernel,
-                                      {{"params", enum_params(inputs.size(), "void * private_p")},
+                                      {{"op_names", op_name_string},
```
Comment (pfultz2) [inline src/targets/gpu/jit/pointwise.cpp:158]: Just name the key `kernel` such as `{"kernel", v.get("kernel", "kernel")}`.

File: src/targets/gpu/jit/pointwise.cpp [hunk 2]
```diff
@@ -167,8 +187,11 @@ struct pointwise_compiler : compiler<pointwise_compiler>
         auto name = g.create_function(
             g.generate_module(*pm).set_attributes({"__device__"}).set_generic_types(*pm));
         std::string lambda = "MIGRAPHX_LIFT(" + name + ")";
+        auto op_names      = get_op_names(*pm);
```
Comment (pfultz2) [inline src/targets/gpu/jit/pointwise.cpp:190]: Do the `join_strings` here and pass the kernel name in.

File: src/targets/gpu/jit/pointwise.cpp (hunk 1 above)
Comment (pfultz2) [reply src/targets/gpu/jit/pointwise.cpp:158]: And set the kernel name with `options.kernel_name = v.get("kernel", "kernel")` with the first `"kernel"` being the key and the second is the fallback value if its missing.

The reason for the fallback is that we may not pass this in when running from the gpu-driver.

File: src/targets/gpu/jit/pointwise.cpp (hunk 2 above)
Comment (kahmed10) [reply src/targets/gpu/jit/pointwise.cpp:184]: fixed

**Review by pfultz2 — APPROVED** [review]

**Review by turneram — APPROVED** [review]

**Pull request merged**